Libraries/src/Amazon.Lambda.AspNetCoreServer/APIGatewayProxyFunction.cs

```diff
@@ -37,6 +39,16 @@ public abstract class APIGatewayProxyFunction
         public bool EnableResponseLogging { get; set; }
 
         /// <summary>
+        /// If true then all response content will be Base64-encoded before being returned.
+        /// </summary>
+        public bool EnableBinaryResponseContent { get; set; }
+
+        /// <summary>
+        /// An optional set of Content-Types that will trigger encoding the response content in Base64.
+        /// </summary>
+        public IEnumerable<string> BinaryResponseContentTypes { get; set; }
```

Comment:
I like the code changes except I would keep _responseEncodingForContentType private at least for now. I'd rather make it protected later if we find we really need to.

Hopefully the API Gateway setting will be made available in the serverless.template at some point which is where I would expect that to be maintain. Right now that seems beyond the scope of this change.